### ISSUE TITLE
Transparency issues when printing

### ISSUE THREAD

**Issue by gioman** [3liz/lizmap-web-client#749]
Setting some level of transparency in layers seems to cause issues when printing a map (not sure of is a LM issue or a QGIS Server one):

1) for rasters: transparency set in LM is not respected at all when printing (is always used the original transparency that comes with the QGIS project(. If transparency is set in QGIS project then printing is ok.

2) for vectors: after setting some transparency in LM and printing then in the result it seems that the transparency of features is still the original and it seems that where there are no features a "translucid" layer has been added. See following images

original (no transparency added in LM)

![screenshot_20180123_111318](https://user-images.githubusercontent.com/1951107/35273031-f51ef4fc-002e-11e8-8707-602a70282c92.png)

transparency added in LM

![screenshot_20180123_111333](https://user-images.githubusercontent.com/1951107/35273035-fdf5ab70-002e-11e8-9aa0-ed690f342b82.png)

testing on QGIS server 2.18.16 and LM 3.1.5

**Comment by josemvm** [comment]
hi @gioman 

i'm using QGIS server 2.14.21, LWC 3.2pre and everything works fine. maybe a QGIS server 2.18 regression...

**Comment by gioman** [comment]
@josemvm @rldhont I also tested on QGIS Server 2.14.* (same QGIS Project and LWC version as in the original description of this issue) and it is confirmed that it works as expected.

**Comment by gioman** [comment]
@mdouchin @rldhont I have understand the reason for this problem:

in QGIS 2.18 the list order of layers in the "LAYERS" parameter of a GetPrint is different compared to QGIS 2.14, but the values for the "OPACITIES" are passed in the same order.

Ex:
2.14

&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetPrint&FORMAT=pdf&EXCEPTIONS=application/vnd.ogc.se_inimage&TRANSPARENT=true&SRS=EPSG:3857&DPI=100&TEMPLATE=land_a4&map0:extent=-1922295.314755,3842163.030118,-1854293.679506,3887914.130304&map0:scale=250000&map0:LAYERS=freguesias,bing_aerial&map0:STYLES=default,default&LAYERS=freguesias,bing_aerial&STYLES=default,default&OPACITIES=153,255

2.18:
&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetPrint&FORMAT=pdf&EXCEPTIONS=application/vnd.ogc.se_inimage&TRANSPARENT=true&SRS=EPSG:3857&DPI=100&TEMPLATE=land_a4&map0:extent=-1922209.719472,3839998.885891,-1854208.084223,3885749.986077&map0:scale=250000&map0:LAYERS=bing_aerial,freguesias&map0:STYLES=default,default&LAYERS=bing_aerial,freguesias&STYLES=default,default&OPACITIES=255,153

in the 2.18 example changing OPACITIES to 

OPACITIES=153,255

"fixes" the problem.

**Comment by signedav** [comment]
Have you had a solution on that @gioman ? From what version on?

**Comment by signedav** [comment]
It works when I change the configuration to <= 2.14 - though I have a QGIS server 2.18.

**Comment by signedav** [comment]
I'm not sure if I need to open a new issue or this one can be reopened?

Sorry to  "reactivating" this, but my problem pretty much matches to this. But the weird thing is, that it works with the wrong configuration and with the correct configuration it does not work.

@gioman I already described it to you directly. But possibly someone else knows the sollution.

Situation is, that with my Lizmap 3.1.4 and QGIs Server 2.18.25 I have the problem, that the oppacity of the layers in the `GetPrint` command does not match with the names of the layers. Means when I set the opacity of OpenStreetMaps to 51, with this request:
```
http://qgis:8999/websig/lizmap/www/index.php/lizmap/service/?repository=main&project=0_visu&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetPrint&FORMAT=pdf&EXCEPTIONS=application/vnd.ogc.se_inimage&TRANSPARENT=true&SRS=EPSG:2056&DPI=100&TEMPLATE=Karte_A4_Hochformat&map0:extent=2679555.4927883143,1179821.667135248,2754557.296371686,1299324.5408447524&map0:scale=500000&map0:LAYERS=OpenStreetMap,Schienennetz,Strecken&map0:STYLES=default,default,default&LAYERS=OpenStreetMap,Schienennetz,Strecken&STYLES=default,default,default&OPACITIES=255,255,51&karten_titel=Titel&karten_untertitel=Untertitel
Like I turn the oppacity of OpenStreetMap to 51:
```
The layers are:
```
LAYERS=OpenStreetMap,Schienennetz,Strecken
```
and the opacities:
```
OPACITIES=255,255,51
```
Not the OpenStreetMap has a low opacity but the "Strecken" instead.

There is a sollution, turning in Lizmap Web Config the `QGIS Server Version` to `<=2.14`. With that it's fine:
Layers:
```
LAYERS=Strecken,Schienennetz,OpenStreetMap
```
match with opacities:
```
OPACITIES=255,255,51
```

But I don't know the influence to other functionalities with the config `<=2.14`. Since I have QGIS Server 2.18.25 I'm a little bit afraid to change that on the customers installation. Or don't I need to be?
There are such things decided regarding the version: https://github.com/3liz/lizmap-web-client/blob/master/lizmap/modules/lizmap/classes/lizmapWFSRequest.class.php#L466
